x/mongo/driver/crypt.go

```diff
@@ -382,3 +389,69 @@ func (c *crypt) decryptKey(kmsCtx *mongocrypt.KmsContext) error {
 		}
 	}
 }
+
+// needsKmsProvider returns true if provider was initially set to an empty document.
+// An empty document signals the driver to fetch credentials.
+func needsKmsProvider(kmsProviders *bsoncore.Document, provider string) bool {
+	gcpVal, err := kmsProviders.LookupErr(provider)
+	if err != nil {
+		// KMS provider is not configured.
+		return false
+	}
+	gcpDoc, ok := gcpVal.DocumentOK()
+	if !ok || len(gcpDoc) != 5 {
+		// KMS provider is not an empty document.
+		return false
+	}
+	return true
```

Comment:
Can return a boolean expression instead of using a conditional block.

E.g.
```go
// KMS provider is an empty document.
return ok && len(providerDoc) == 5
```